File: lista 2/7.py
a, b = map(int, input().split())
c, d = map(int, input().split())

area_tri = (a * b) / 2.0
area_ret = float(c * d)

cabe_direto = (a <= c and b <= d)
cabe_girado = (a <= d and b <= c)

possivel = (cabe_direto or cabe_girado) and (area_tri <= area_ret)

if possivel:
    sobra = area_ret - area_tri
    print("Ao infinito e alem.")
    print(f"{area_tri:.1f} {area_ret:.1f} {sobra:.1f}")
else:
    print("Vamos treinar na caverna do Superman.")
    print(f"{area_tri:.1f} {area_ret:.1f} {area_ret:.1f}")

File: lista 2/7.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main() {
    double a, b, c, d;
    cin >> a >> b >> c >> d;

    double area_tri = (a * b) / 2.0;
    double area_ret = c * d;

    bool cabe_direto = (a <= c && b <= d);
    bool cabe_girado = (a <= d && b <= c);

    bool possivel = (cabe_direto || cabe_girado) && (area_tri <= area_ret);

    if (possivel) {
        double sobra = area_ret - area_tri;
        cout << "Ao infinito e alem." << endl;
        cout << fixed << setprecision(1) << area_tri << " " << area_ret << " " << sobra << endl;
    } else {
        cout << "Vamos treinar na caverna do Superman." << endl;
        cout << fixed << setprecision(1) << area_tri << " " << area_ret << " " << area_ret << endl;
    }

    return 0;
}